packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -643,60 +644,142 @@ export class SDKClient {
   private createFile = async (
     callData: Uint8Array,
     client: Client,
-    requestId?: string,
-    callerName?: string,
-    interactingEntity?: string,
+    requestId: string,
+    callerName: string,
+    interactingEntity: string,
   ) => {
     const requestIdPrefix = formatRequestIdMessage(requestId);
     const hexedCallData = Buffer.from(callData).toString('hex');
+    const currentDateNow = Date.now();
+    let createFileId, fileCreateTx, fileAppendTx;
 
-    const fileCreateTx = await new FileCreateTransaction()
-      .setContents(hexedCallData.substring(0, this.fileAppendChunkSize))
-      .setKeys(client.operatorPublicKey ? [client.operatorPublicKey] : []);
-    const fileCreateTxResponse = await fileCreateTx.execute(client);
-    const { fileId } = await fileCreateTxResponse.getReceipt(client);
-
-    const createFileRecord = await fileCreateTxResponse.getRecord(this.clientMain);
-    this.captureMetrics(
-      SDKClient.transactionMode,
-      fileCreateTx.constructor.name,
-      Status.Success,
-      createFileRecord.transactionFee.toTinybars().toNumber(),
-      createFileRecord?.contractFunctionResult?.gasUsed,
-      callerName,
-      interactingEntity,
-    );
+    try {
+      const shouldLimit = this.hbarLimiter.shouldLimit(currentDateNow, SDKClient.transactionMode, callerName);
+      if (shouldLimit) {
+        throw predefined.HBAR_RATE_LIMIT_EXCEEDED;
+      }
 
-    if (fileId && callData.length > this.fileAppendChunkSize) {
-      const fileAppendTx = await new FileAppendTransaction()
-        .setFileId(fileId)
-        .setContents(hexedCallData.substring(this.fileAppendChunkSize, hexedCallData.length))
-        .setChunkSize(this.fileAppendChunkSize)
-        .setMaxChunks(this.maxChunks);
-      await fileAppendTx.execute(client);
+      const fileCreateTx = await new FileCreateTransaction()
+        .setContents(hexedCallData.substring(0, this.fileAppendChunkSize))
+        .setKeys(client.operatorPublicKey ? [client.operatorPublicKey] : []);
+
+      const fileCreateTxResponse = await fileCreateTx.execute(client);
+      const { fileId } = await fileCreateTxResponse.getReceipt(client);
+
+      // get transaction fee and add expense to limiter
+      const createFileRecord = await fileCreateTxResponse.getRecord(this.clientMain);
+      let transactionFee = createFileRecord.transactionFee;
+      this.hbarLimiter.addExpense(transactionFee.toTinybars().toNumber(), currentDateNow);
 
       this.captureMetrics(
         SDKClient.transactionMode,
-        fileAppendTx.constructor.name,
+        fileCreateTx.constructor.name,
         Status.Success,
-        await this.calculateFileAppendTxTotalTinybarsCost(fileAppendTx),
-        0,
+        createFileRecord.transactionFee.toTinybars().toNumber(),
+        createFileRecord?.contractFunctionResult?.gasUsed,
```

Comment:
make this 0, no gas is utilized on FileCreate/FileAppend

Reply:
Agreed, but I'll do this in a separate issue as it is unrelated to the HBar spend and has existed there for a while.